PD-1714: Update colors in palette to reflect brand refresh (#1080)
mongodb/leafygreen-ui: packages/palette/UPGRADE.md

# v3 to v4

> v3.3 and above of the palette package contains the updated palette for the brand refresh. In v4, the old colors are removed entirely from this package.

Reviewer: ```suggestion
v3.3 and above of the palette package contains the updated palette for the brand refresh. In v4, the old palette is removed entirely from this package.
```

Author: Good call.

## Imports

### JS

The named export for the new palette has changed from `uiColors` to `palette`. You'll need to import the new colors like so:

```js
import { palette } from '@leafygreen-ui/palette';
```

### Less

The name of the included `.less` variables file has changed from 'ui-colors.less' to `palette.less`. You'll need to import the new file like so:

```less
@import '<path to node_modules>/@leafygreen-ui/palette/dist/palette.less';
```

## Color Variables

### JS

The name and structure of the JS variables is largely unchanged. Here are the notable changes:

- The `focus` variable has been removed. It's recommended to update usage of this color to `palette.blue.light1`.
- The `palette.red.dark1` variable has been removed. It's recommended to update usage of this color to `palette.red.dark2`.
- We've added a new `palette.green.light1` color.
- We've added a new `palette.blue.dark2` color.
- We've added a new `palette.red.light1` color.
- We've added purple to the palette. You can find `dark3`, `dark2`, `base`, `light2`, and `light3` values for the color.

### Less

There's a number of notable changes to the Less variables that are available.

> - All variables have moved from a `@leafygreen__[color name]` format to a `@palette__[color name]` format.

Reviewer: Should we prefix so it's `@lgpalette`? It looks gross but is a nice sandbox. But maybe too overtly defensive.

Author: I was trying to distance the new palette from being "leafygreen", since it's more broad than that (though yes, the package itself is leafygreen). Given the specific BEM-like namespacing of these colors, I was thinking the likelihood of collisions is extremely minute. Do you think that's incorrect?

Reviewer: Oh that's fair. Also requires a consumer to import the LESS file. Yeah, leave as is!

- The `@leafygreen__focus` variable has been removed. It's recommended to update usage of this color to `@palette__blue--light-1`.
- The `@leafygreen__red--dark-1` variable has been removed. It's recommended to update usage of this color to `@palette__red--dark-2`.
- We've added a new `@palette__green--light-1` color.
- We've added a new `@palette__blue--dark-2` color.
- We've added a new `@palette__red--light-1` color.
- We've added purple to the palette. You can find `dark-3`, `dark-2`, `base`, `light-2`, and `light-3` values for the color.

## Color Changes

Nearly every color in the palette has been changed to some extent. You'll notice colors are more vibrant all around, though many are compatible with the usage of the previous colors. Here are the most notable changes:

> - Base Green is lighter, and more vibrant than the previous Base Green. For a 1:1 update, we recommend changing instances of that green to Green: Dark 1.

Reviewer: Can we inline an example using the JS?

Author: We can, though I was trying to be language-agnostic here. Does that matter?

Reviewer: meh. Can give an example for both, as it's def gonna be a common thing referenced.

Author: Fair enough. I'll update 👍

Author: How's this update look?

> - Base Green is lighter, and more vibrant than the previous Base Green. For a 1:1 update, we recommend changing instances of that green to `palette.green.dark1`/`@palette__green--dark-1`.

Reviewer: ```suggestion
- Base Green (`uiColors.green.base` / `@leafygreen__green--base`) is lighter, and more vibrant than the previous Base Green. For a 1:1 update, we recommend changing instances of that green to `palette.green.dark1`/`@palette__green--dark-1`.
```

To make the copy pasta even easier!
